### ISSUE TITLE
When a devfile references a parent we should not add the default component

### ISSUE THREAD

**Issue by l0rd** [eclipse/che#22103]
### Is your enhancement related to a problem? Please describe

When a devfile has parent ([full sample](https://github.com/l0rd/inventory-quarkus/blob/parent-by-id/.devfile.yaml)):

```yaml
schemaVersion: 2.2.0
metadata:
  name: inventory-quarkus
parent:
  id: java-maven
  registryUrl: https://registry.devfile.io/
  version: 1.2.0
```

the resulting Pod has 3 containers (the gateway, the one defined in the parent and the default one):

```bash
$ kubectl get pods <pod-name> -o json | jq -r '.spec.containers[].image'
registry.access.redhat.com/ubi8/openjdk-11:latest
quay.io/devfile/universal-developer-image:ubi8-38da5c2
quay.io/eclipse/che--traefik:v2.9.6-bb7be8d50edf73d8d3a812ac8873ef354a0fe9b40d7f3880747b43a3525855d2
```

### Describe the solution you'd like

The default component (the `universal-developer-image` in the case above) should not be added if there is a parent.

**Comment by azatsarynnyy** [comment]
I'm not sure if it's related to this bug, but when I try to create a workspace from the project https://github.com/l0rd/inventory-quarkus/tree/parent-by-id, the Dashboard failed to create it on our dogfooding instance (7.64.0-next):
![image](https://user-images.githubusercontent.com/1636395/228469801-7a864265-12f2-4e30-a05c-5d9485f004a4.png)
Clicking `Continue ...` doesn't change anything.

However, the workspace is created without any problems on the DevSandbox instance where 7.60.2 is deployed.

**Comment by ibuziuk** [comment]
would be nice to backport it to Dev Spaces 3.5

**Comment by l0rd** [comment]
Now I cannot even start a workspace

<img width="948" alt="image" src="https://user-images.githubusercontent.com/606959/232733872-78c9468b-0fce-489b-be8c-f0caa3ec7435.png">

**Comment by ibuziuk** [comment]
Reopening, reproduced the same issue with https://github.com/che-incubator/quarkus-api-example/tree/parent that references https://registry.stage.devfile.io/viewer/devfiles/community/udi

**Comment by ibuziuk** [comment]
We investigated the issue with @olexii4 and it looks like the problem is on the devworkspace generator end cc: @svor 

Basically, when devfile contains the parent reference we should not fail with the error. 
N.B It is important to backport the fix to 7.64.x for Dev Spaces 3.6 cc: @nickboldt

**Comment by ibuziuk** [comment]
@l0rd should be fixed and also backported to 7.64.x for 3.6
verifed on dogfooding against https://github.com/che-incubator/quarkus-api-example/pull/40

**Comment by ibuziuk** [comment]
we can technically introduce another endpoint on the UD end that will do the `podman login` to the internal OpenShift registry and call it during the workspace startup

![image](https://user-images.githubusercontent.com/1461122/233628682-3dd4dc7e-e2a1-48fb-8af5-06a1c48bbccc.png)

Draft PR -  https://github.com/eclipse-che/che-dashboard/pull/785
